src/algebra/homology/additive.lean

```diff
@@ -0,0 +1,236 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.homology.homology
+import algebra.homology.single
+import category_theory.preadditive.additive_functor
+
+/-!
+# Homology is an additive functor
+
+When `V` is preadditive, `homological_complex V c` is also preadditive,
+and `homology_functor` is additive.
+
+TODO: similarly for `R`-linear.
+-/
+
+universes v u
+
+open_locale classical
+noncomputable theory
+
+open category_theory category_theory.limits homological_complex
+
+variables {ι : Type*}
+variables {V : Type u} [category.{v} V] [preadditive V]
+
+variables {c : complex_shape ι} {C D E : homological_complex V c}
+variables (f g : C ⟶ D) (h k : D ⟶ E) (i : ι)
+
+namespace homological_complex
+
+instance : has_zero (C ⟶ D) := ⟨{ f := λ i, 0 }⟩
+instance : has_add (C ⟶ D) := ⟨λ f g, { f := λ i, f.f i + g.f i, }⟩
+instance : has_neg (C ⟶ D) := ⟨λ f, { f := λ i, -(f.f i), }⟩
+instance : has_sub (C ⟶ D) := ⟨λ f g, { f := λ i, f.f i - g.f i, }⟩
+
+@[simp] lemma zero_f_apply (i : ι) : (0 : C ⟶ D).f i = 0 := rfl
+@[simp] lemma add_f_apply (f g : C ⟶ D) (i : ι) : (f + g).f i = f.f i + g.f i := rfl
+@[simp] lemma neg_f_apply (f : C ⟶ D) (i : ι) : (-f).f i = -(f.f i) := rfl
+@[simp] lemma sub_f_apply (f g : C ⟶ D) (i : ι) : (f - g).f i = f.f i - g.f i := rfl
+
+instance : add_comm_group (C ⟶ D) :=
```

Comment:
```suggestion

/- TODO(jmc/Scott): the instance below doesn't have the correct defeq for `nsmul` and `gsmul`.
We should generalize `function.injective.add_comm_group` and friends.
For the `R`-linear version, it will be very convenient to have a good definition of `nsmul` and `gsmul`,
that matches `smul`. -/

instance : add_comm_group (C ⟶ D) :=
```